src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOf.cs

```diff
@@ -0,0 +1,227 @@
+                                            if (variableInitializer.Value is IInvocationOperation invocationOperation)
+                                            {
+                                                if (InvocationOperationHasTargetMethod(invocationOperation))
+                                                {
+                                                    var locations = ImmutableArray.Create(blockLocation, declarationGroupLocation);
+                                                    variableNameToLocationsMap.TryAdd(variableDeclaratorOperation.Symbol.Name, locations);
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                            else if (leftOperand is IInvocationOperation leftInvocationOperation)
+                            {
+                                var blockLocationList = new List<Location> { blockLocation };
+                                if (InvocationOperationHasTargetMethod(leftInvocationOperation))
+                                {
+                                    leftOperandInvocationLocations = blockLocationList;
+                                }
+                            }
+                        }
+                    },
+                    OperationKind.Binary);
+
+                    context.RegisterOperationBlockEndAction(context =>
+                    {
+                        if (leftOperandInvocationLocations != null)
+                        {
+                            context.ReportDiagnostic(leftOperandInvocationLocations.CreateDiagnostic(Rule));
+                        }
+                        if (variableNameToLocationsMap.Count > 0)
+                        {
+                            List<Location> locations = new List<Location>();
+                            foreach (KeyValuePair<string, ImmutableArray<Location>> variableNameAndLocation in variableNameToLocationsMap)
+                            {
+                                string variableName = variableNameAndLocation.Key;
+                                if (!variableNameToNumberOfReferences.TryGetValue(variableName, out int numberOfReferences))
+                                {
+                                    continue;
+                                }
+
+                                if (numberOfReferences > 1)
+                                {
+                                    continue;
+                                }
+
+                                context.ReportDiagnostic(variableNameAndLocation.Value.CreateDiagnostic(Rule));
+                            }
+                        }
+                    });
+                });
+
+                bool InvocationOperationHasTargetMethod(IInvocationOperation invocationOperation)
```

Comment:
Is there a benefit on having this method declared locally inside `Initialize`?

Reply:
Yup, I put it there to re-use some vars declared above. I'll change it to a non-local method to improve readability